### DISCUSSION TITLE
I need a C# library similar to rust-obfstr

### DISCUSSION THREAD

**Discussion by dataangel** [dotnet/runtime#96925] [Ideas]
This way, the text is secure.

**Comment by Clockwork-Muse** [comment]
It's vanishingly unlikely the runtime would provide such a library, when it can be provided by a third-party package (as it was with rust).

That said, the _only_ thing such a package provides is trivial string extraction by just reading the hex of the binary.  The actual data is and must still be recoverable (because otherwise the application couldn't use it), although it may be slightly more difficult.
If you need to keep something actually secure, you **cannot** distribute it with your binary, and likely need to retain it inside your own service.
